tests/utils/config_test.py

```diff
@@ -0,0 +1,165 @@
+import os
+
+from unittest import TestCase
+
+from esrally import exceptions
+from esrally.utils import config
+
+class ConfigHelperFunctionTests(TestCase):
+    def test_csv_to_list(self):
+        self.assertEqual([], config.csv_to_list(""))
+        self.assertEqual(["a", "b", "c", "d"], config.csv_to_list("    a,b,c   , d"))
+        self.assertEqual(["a-;d", "b", "c", "d"], config.csv_to_list("    a-;d    ,b,c   , d"))
+
+    def test_kv_to_map(self):
+        self.assertEqual({}, config.kv_to_map([]))
+        self.assertEqual({"k": "v"}, config.kv_to_map(["k:'v'"]))
+        self.assertEqual({"k": "v", "size": 4, "empty": False, "temperature": 0.5},
+                         config.kv_to_map(["k:'v'", "size:4", "empty:false", "temperature:0.5"]))
+
+
+class TestTargetHosts(TestCase):
+    def test_empty_arg_parses_as_empty_list(self):
+        self.assertEqual([], config.TargetHosts('')())
+        self.assertEqual([], config.TargetHosts('').default)
+        self.assertEqual({'default': []}, config.TargetHosts('').all_hosts)
+
+    def test_csv_hosts_parses(self):
+        target_hosts = '127.0.0.1:9200,10.17.0.5:19200'
+
+        self.assertEqual(
+            {'default': [{'host': '127.0.0.1', 'port': 9200},{'host': '10.17.0.5', 'port': 19200}]},
+            config.TargetHosts(target_hosts).all_hosts
+        )
+
+        self.assertEqual(
+            [{'host': '127.0.0.1', 'port': 9200},{'host': '10.17.0.5', 'port': 19200}],
+            config.TargetHosts(target_hosts).default
+        )
+
+        self.assertEqual(
+            [{'host': '127.0.0.1', 'port': 9200},{'host': '10.17.0.5', 'port': 19200}],
+            config.TargetHosts(target_hosts)()
+        )
+
+    def test_jsonstring_parses_as_dict_of_clusters(self):
+        target_hosts = '{"default": ["127.0.0.1:9200","10.17.0.5:19200"], "remote_1": ["88.33.22.15:19200"], "remote_2": ["10.18.0.6:19200","10.18.0.7:19201"]}'
+
+        self.assertEqual(
+            {'default': ['127.0.0.1:9200','10.17.0.5:19200'],
+             'remote_1': ['88.33.22.15:19200'],
+             'remote_2': ['10.18.0.6:19200','10.18.0.7:19201']},
+            config.TargetHosts(target_hosts).all_hosts)
+
+    def test_json_file_parameter_parses(self):
+        self.assertEqual(
+            {"default": ["127.0.0.1:9200","10.127.0.3:19200"] },
+            config.TargetHosts(os.path.join(os.path.dirname(__file__), "resources/target_hosts_1.json")).all_hosts)
```

Comment:
Nit: I think this should rather be `os.path.join(os.path.dirname(__file__), "resources", "target_hosts_1.json")`) (all path components as separate elements)

Reply:
Yes makes perfect sense. Addressing.